scripts/check_branch.py

```diff
@@ -50,7 +81,7 @@ def check_branch(branch: str | None, deployment: str) -> None:
             raise BranchDeploymentMismatch(branch, deployment, deployments)
 
 
-def target_branch() -> str | None:
+def target_branch() -> tuple[str | None, BranchContext | None]:
```

Comment:
The fact that there is a fourth case (`BranchContext is None`) in addition to the cases that `BranchContext` specifies explicitly, is a bit of a smell. In combination with the fact that `branch` can also be `None` makes things even more complex.